packages/yarnpkg-shell/tests/shell.test.ts

```diff
@@ -1566,4 +1562,35 @@ describe(`Shell`, () => {
       });
     });
   });
+
+  describe(`Background jobs`, () => {
+    it(`should provide color-coded prefixed output and color-coded "Job has ended" message inside TTYs`, async () => {
```

Comment:
Can you add a test doing an actual `sleep` (perhaps implemented as a builtin in tests only)? This way we'd also guard against race conditions like the one there used to be.